lib/manageiq/providers/ibm_cloud/cloud_tools/common/proxy.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        module Common
+          # Set proxy configurations dynamically.
+          class Proxy
+            def initialize
+              @keys = %i[address port username password headers]
+              @keys.each { |key| self.class.attr_accessor(key) }
+            end
+
+            # Add a server configuration.
+            # @return [Proxy]
+            def server(address:, port:)
+              @address = address
+              @port = port
+              self
+            end
+
+            # Add a proxy authentication to the server configuration.
+            # @return [Proxy]
+            def authentication(username:, password:)
+              @username = username
+              @password = password
+              self
+            end
+
+            # Convert non-nil keys into a hash.
+            # @return [Hash]
+            def to_hash
+              data = {}
+              @keys.each do |key|
+                variable = instance_variable_get("@#{key}")
```

Comment:
This can end up exposing passwords.  You should at least hardcode password as a key to skip over.